samza-core/src/main/java/org/apache/samza/operators/util/InternalInMemoryStore.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.operators.util;
+
+import org.apache.samza.storage.kv.Entry;
+import org.apache.samza.storage.kv.KeyValueIterator;
+import org.apache.samza.storage.kv.KeyValueStore;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implements a {@link KeyValueStore} using an in-memory Java Map.
+ * @param <K>
+ * @param <V>
+ *
+ * TODO: Provide a persistent implementation of the KeyValueStore API. Maybe, we don't need this class.
```

Comment:
This is also because that we don't have a way to instantiate the KeyValueStore w/o the configuration, right? The TaskContext does not support creating a pluggable KeyValueStore from within the program. Hence, we have to create this class as a close-gap solution in short-term. We should make this point clear as well.